### PR TITLE
let convert accept dom.Node as input

### PR DESCRIPTION
**Pull request by jarontai**
issue: https://github.com/jarontai/html2md/issues/29

### CODE REVIEW

File: lib/src/converter.dart [hunk 1]
```diff
@@ -36,14 +37,18 @@ final _trailingNewLinesRegExp = RegExp(r'\n*$');
 /// The [rules] parameter can be used to customize element processing.
 ///
 String convert(
-  String html, {
+  dynamic input, {
```
Comment (lindeer) [inline lib/src/converter.dart:40]: maybe `Object` is better, this could be null safety.

File: lib/src/converter.dart [hunk 2]
```diff
@@ -36,14 +37,18 @@ final _trailingNewLinesRegExp = RegExp(r'\n*$');
...
   Map<String, String>? styleOptions,
   List<String>? ignore,
   List<Rule>? rules,
 }) {
-  if (html.isEmpty) {
+  // the input can be html string or dom.Node
+  if (input is! String && input is! dom.Node) {
+    return '';
+  }
+  if (input is String && input.isEmpty) {
```
Comment (lindeer) [inline lib/src/converter.dart:51]: merge into one condition? `if (input is String && input.isEmpty || input is! dom.Noe)`

File: lib/src/node.dart [hunk 3]
```diff
@@ -18,13 +18,17 @@ class Node {
...
+  factory Node.root(dynamic input, {String? rootTag}) {
+    dom.Node? root;
+    if (input is String) {
+      var doc = parse(input);
+      if (rootTag != null && rootTag.isNotEmpty) {
+        root = doc.getElementsByTagName(rootTag).first;
+      }
+      root ??= doc.getElementsByTagName('html').first;
+    } else {
+      root = input as dom.Node;
```
Comment (lindeer) [inline lib/src/node.dart:30]: what if html fragment? e.g.:
```
<p>some thing</p>
<div>one another thing</div>
```

**Comment by jarontai** [comment]
@lindeer Thank you for your review, can you give some help :sweat_smile:, what I mean is that could you help to fix the issues you raised? I didn't use dart/flutter in my work, I haven't written dart code for a long time.

**Comment by lindeer** [comment]
that's fine, but I do not have permission to push `dynamic_input` branch

**Comment by jarontai** [comment]
maybe you can checkout this branch and create a new branch (PR) based on it

**Pull request closed without merging**